cleanlab/internal/multilabel_utils.py

```diff
@@ -0,0 +1,255 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Helper classes and functions used internally to compute label quality scores in multi-label classification.
+"""
+
+from enum import Enum
+import itertools
+from typing import Callable, Optional
+
+import numpy as np
+from sklearn.model_selection import cross_val_predict
+from sklearn.utils.multiclass import is_multilabel
```

Comment:
FYI depending on sklearn.utils is not ideal:
https://scikit-learn.org/stable/developers/utilities.html#developers-utils

> Warning These utilities are meant to be used internally within the scikit-learn package. They are not guaranteed to be stable between versions of scikit-learn. Backports, in particular, will be removed as the scikit-learn dependencies evolve.

Reply:
so consider just implementing our own version of this basic function as needed in this file.